Tests/PenguinGraphTests/GraphCopyingTests.swift

```diff
@@ -111,9 +111,40 @@ final class GraphCopyingTests: XCTestCase {
     XCTAssertEqual(2, dst.inDegree(of: 2))
   }
 
+  func testAddingEdges() {
+    var src = DirectedAdjacencyList<String, String, Int>()
+    _ = src.addVertex()
+    _ = src.addVertex()
+    _ = src.addVertex()
+
+    _ = src.addEdge(from: 1, to: 2, storing: "1->2 (src)")
```

Comment:
IMO it's poor form to write these tests assuming you know how vertex IDs are going to be allocated or even that they're going to be `Int`s.  That's not what you're trying to test here, and if the implementation changes, the test will break for reasons unrelated to what it's trying to test.

That's not to mention the fact that this isn't how people are going to use the library.  You have an opportunity here to see what it's like to write real code with your API, but it's somewhat squandered.